pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -992,9 +1043,138 @@ func (r *ReconcileClusterPool) createCloudBuilder(pool *hivev1.ClusterPool, logg
 		cloudBuilder.Region = platform.Azure.Region
 		cloudBuilder.CloudName = platform.Azure.CloudName
 		return cloudBuilder, nil
-	// TODO: OpenStack, VMware, and Ovirt.
+	case platform.OpenStack != nil:
+		credsSecret, err := r.getCredentialsSecret(pool, platform.OpenStack.CredentialsSecretRef.Name, logger)
+		if err != nil {
+			return nil, err
+		}
+		cloudBuilder := clusterresource.NewOpenStackCloudBuilderFromSecret(credsSecret)
+		cloudBuilder.Cloud = platform.OpenStack.Cloud
+		return cloudBuilder, nil
+	// TODO: VMware, and Ovirt.
 	default:
 		logger.Info("unsupported platform")
 		return nil, errors.New("unsupported platform")
 	}
 }
+
+func (r *ReconcileClusterPool) getInventoryCustomization(pool *hivev1.ClusterPool, logger log.FieldLogger) (*corev1.LocalObjectReference, error) {
+	cdcRefs := pool.Spec.Inventory.ClusterDeploymentCustomizations
+	for _, ref := range cdcRefs {
+		cdc := &hivev1.ClusterDeploymentCustomization{}
+		refNamespacedName := types.NamespacedName{Namespace: pool.Namespace, Name: ref.Name}
+		// TODO: context should be TODO or Background?
+		// I'm not sure if this will work
+		err := r.Get(context.TODO(), refNamespacedName, cdc)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logger.Infof("customization not found: %s", ref.Name)
+			}
+			log.WithError(err).Error("error reading customization")
+		}
+
+		// TODO: kinda fugly
+		currentAvailability := controllerutils.FindClusterDeploymentCustomizationCondition(
+			cdc.Status.Conditions,
+			hivev1.ClusterDeploymentCustomizationAvailableCondition,
+		)
+		// Customization available
+		if currentAvailability.Status == corev1.ConditionFalse {
+			r.setInventoryValidCondition(pool, false, logger)
+			continue
+		}
```

Comment:
Why do we set a `InventoryValid` condition when an entry is unavailable? IIUC, this condition is only set when an entry is invalid/malformed or when it is mentioned in the inventory but actual CR does not exist

Reply:
I misunderstood the following line in the design (https://github.com/openshift/hive/blob/master/docs/enhancements/clusterpool-inventory.md?plain=1#L97): 
> - If the ClusterDeploymentCustomization is unavailable, we set `InventoryValid` condition to false on ClusterPool indicating that some entries in the inventory are invalid or missing.